### DISCUSSION TITLE
Sizing Tiles of Patterns Adaptively

### DISCUSSION THREAD

**Discussion by blue-j** [ImageMagick/ImageMagick#4576] [Help] [answered]
I'd love to be able to control the size of the checkerboard pattern, so it looks similar across differently sized image.

I'm using

`magick composite -compose Dst_Over -tile pattern:checkerboard  input.png output.png`

Is there any way to control the relative size of the pattern?  Thank you in advance for your time and support!  : )

- 
![sample1](https://user-images.githubusercontent.com/3990142/145666003-d9e8b02a-e6ae-4ef7-8210-cf6c878e1037.jpg)
![sample2](https://user-images.githubusercontent.com/3990142/145666005-c0b04b9b-cc4b-4a37-b4ba-10d0f4f56855.jpg)
J

**Comment by fmw42** [comment]
Don't use magick composite. Use magick. Then resize the checkerboard pattern in parentheses before tiling. Save the resized tile to mpr: and then call it in with tile:mpr:resize_tile.  See tiling in memory at https://legacy.imagemagick.org/Usage/canvas/#tile. If you post your input image, we can suggest a command.

**Comment by fmw42** [comment, marked as answer]
Here is one way to do that:

Transparent Input:

![logot](https://user-images.githubusercontent.com/13458218/145666501-a9fe8390-2874-4fb2-906d-f3ed4cb787bb.png)

```
magick logot.png \
\( pattern:checkerboard -resize 200% -write mpr:tile +delete \) \
\( +clone -fill mpr:tile  -draw 'color 0,0 reset' \) \
+swap -compose over -composite logo_tile.png

```

![logo_tile](https://user-images.githubusercontent.com/13458218/145666516-235e8c32-7b4b-4a4f-aec9-a28070a8332d.png)

**Reply by blue-j** [reply]
Thank you!
